objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+                return tbl.index
+            end
+        end
+    end
+    return nil
+end
+
+function timePasses()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    for _,tbl in pairs(impendTable) do
+        local var = tbl.energy[1] + tbl.energy[2] + tbl.energy[3]
+        if var >= tbl.maxEnergy and tbl.turnSelected ~= Global.getVar("turn") - 1 then
+            clearCardData(tbl.guid)
+        else
+            tbl.energy[1] = var
+            tbl.energy[2] = 0
+            tbl.energy[3] = 0
+            updateEnergyDisplay(tbl.guid)
+        end
+    end
+    g3Selected = true
+    g3Toggle(dances)
+    updateSave()
+end
+
+local function energyPerTurn(dances)
+    local hits = Physics.cast({
+        origin = dances.positionToWorld(energyPoint),
+        direction = Vector(0, 1, 0),
+        max_distance = 1,
+        type = 1, --ray
+        -- debug = true
+    })
+    for _, hit in pairs(hits) do
+        if hit.hit_object.hasTag("Presence") then
+            return 1
+        end
+    end
+    return 2
+end
+
+function onGainPay(params)
+    local dances = Global.call("getSpirit", {name = spiritName})
+    if params.color == Global.call("getSpiritColor", {name = spiritName}) then
+        if params.isGain then
+            for _,tbl in pairs(impendTable) do
+                if params.isUndo then
+                    impendTable[tbl.guid].energy[2] = 0
+                elseif tbl.turnSelected ~= Global.getVar("turn") then
+                    impendTable[tbl.guid].energy[2] = energyPerTurn(dances)
+                end
+                updateEnergyDisplay(tbl.guid)
+            end
+            updateSave()
+        else
+            local zone = Global.getVar("selectedColors")[params.color].zone
+            if params.isUndo then
+                for _,obj in pairs(zone.getObjects()) do
+                    local tbl = impendTable[obj.guid]
+                    if tbl then
+                        -- originally pulled all impend cards back to impend hand but that can be annoying
+                        -- swapping to only pull back cards not currently with sufficient energy
+                        if tbl.energy[1] + tbl.energy[2] + tbl.energy[3] < tbl.maxEnergy then
+                            obj.deal(1, params.color, 3)
+                        end
+                    end
+                end
+            else
+                -- Store all impended cards with enough energy into a table
+                local toPlayFast = {}
+                local toPlaySlow = {}
+                local impendInPlay = {}
+                for _,obj in pairs(zone.getObjects()) do
+                    local tbl = impendTable[obj.guid]
+                    if tbl then
+                        if tbl.energy[1] + tbl.energy[2] + tbl.energy[3] >= tbl.maxEnergy then
+                            impendInPlay[obj] = true
+                        else
+                            obj.deal(1, params.color, 3)
```

Comment:
when would impending cards be in the play area already that shouldn't be there?

Reply:
Quite often people will want to put the cards into the play area to check what elements they have for hitting thresholds rather than manually counting/adding on the impended elements allowing them to decide things like which cards to take. I had intentionally made this system to work for the cards to be placed/moved anywhere such that things like this are possible.

Reply:
gotcha g3 play makes sense